src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -254,3 +371,87 @@ class JaxArrayField(_BaseNdArrayField):
         array_ns: ClassVar[ModuleType] = jnp
 
     common.field.register(jnp.ndarray, JaxArrayField.from_array)
+
+
+def _find_index_of_dim(dim: Dimension, domain_slice: DomainSlice) -> Optional[int]:
+    for i, (d, _) in enumerate(domain_slice):
+        if dim == d:
+            return i
+    return None
+
+
+def broadcast(field: common.Field, new_dimensions: tuple[common.Dimension, ...]):
```

Comment:
```suggestion
def _broadcast(field: common.Field, new_dimensions: tuple[common.Dimension, ...]):
```
sucht that users use `fbuiltins.broadcast` and not accidentally this one